libusb/os/windows_winusb.c

```diff
@@ -3860,7 +3860,9 @@ static int hid_open(int sub_api, struct libusb_device_handle *dev_handle)
 
 		priv->hid->string_index[1] = dev->device_descriptor.iProduct;
 		if (priv->hid->string_index[1] != 0)
-			HidD_GetProductString(hid_handle, priv->hid->string[1], sizeof(priv->hid->string[1]));
+			// Use HidD_GetIndexedString() instead of HidD_GetProductString() to get the correct iProduct string whenever
```

Comment:
This sounds like a comment for the commit message, but is likely confusing in the code itself since it refers to previous code that is not present any longer.

Reply:
I expanded the comment somewhat. Better?

Don't know what to do about the commit message? Will this be a squash merge, where the message can be edited? Suggestions?

Reply:
Thanks, that seems better to me. Yes, we will always squash such commits, and you can also squash and repush yourself.

Since this is non-obvious, you can also add the description of the observed issue to the commit message.

Reply:
OK, would you like me to do the squash? Not sure about the repush... I guess I would have to force push in order to overwrite the existing branch, would GitHub handle the reference to this issue anyway?

About the commit message; by "add the description" do you mean my updated code comment, or what I wrote in [my issue below](https://github.com/libusb/libusb/pull/1091#issuecomment-1072091703)?

Reply:
Yes, you can simply force push on the same branch. I meant the PR comment below where you describe the observed bug.